crates/rome_js_analyze/src/semantic_analyzers/nursery/no_confusing_arrow.rs

```diff
@@ -0,0 +1,113 @@
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var x = a => 1 ? 2 : 3;
+    /// var x = (a) => 1 ? 2 : 3;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var x = a => (1 ? 2 : 3);
+    /// var x = (a) => (1 ? 2 : 3);
+    /// var x = (a) => {
+    ///     return 1 ? 2 : 3;
+    /// };
+    /// var x = a => { return 1 ? 2 : 3; };
+    /// ```
+    ///
+    pub(crate) NoConfusingArrow {
+        version: "next",
+        name: "noConfusingArrow",
+        recommended: false,
+    }
+}
+
+impl Rule for NoConfusingArrow {
+    type Query = Semantic<JsArrowFunctionExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let arrow_fn = ctx.query();
+
+        if arrow_fn.fat_arrow_token().is_err() {
+            return None;
+        }
+
+        match arrow_fn.body() {
+            Ok(AnyJsFunctionBody::AnyJsExpression(AnyJsExpression::JsConditionalExpression(_))) => {
+                Some(())
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let arrow_fn = ctx.query();
+
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                arrow_fn
+                    .fat_arrow_token()
+                    .expect("We checked that a `fat_arrow_token` exists in `run`")
+                    .text_trimmed_range(),
+                markup! {
+                    "Ambiguous fat arrow."
+                },
+            )
+            .note(markup! {
+                "Fat arrows are similar to some comparison operators ("
+                    <Emphasis>"<"</Emphasis>", "
+                    <Emphasis>">"</Emphasis>", "
+                    <Emphasis>"<="</Emphasis>", "
+                    <Emphasis>">="</Emphasis>
+                ")."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let body_expr = match ctx.query().body() {
+            Ok(AnyJsFunctionBody::AnyJsExpression(expr)) => expr,
+            _ => unreachable!("We checked that the body is an expression in `run`"),
+        };
+
+        let mut mutation = ctx.root().begin();
+
+        mutation.replace_node(
+            body_expr.clone(),
+            AnyJsExpression::from(make::js_parenthesized_expression(
+                JsSyntaxToken::new_detached(JsSyntaxKind::L_PAREN, "(", [], []),
+                body_expr,
+                JsSyntaxToken::new_detached(JsSyntaxKind::R_PAREN, ")", [], []),
+            )),
+        );
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Wrap body in parenthesis." }.to_owned(),
```

Comment:
I think `the` is missing. I am not sure about that (English is not my native language).

```suggestion
            message: markup! { "Wrap the function body in parenthesis." }.to_owned(),
```